ha/test/integration/health_monitor/test_event_subscriber.py

```diff
@@ -34,15 +36,39 @@ class TestEventManager(unittest.TestCase):
     def setUp(self):
         self.event_manager = EventManager.get_instance()
         self.component = SUBSCRIPTION_LIST.TEST
-        self.event = SubscribeEvent(RESOURCE_TYPES.NODE,
-            [RESOURCE_STATUS.FAILED, RESOURCE_STATUS.ONLINE, RESOURCE_STATUS.DEGRADED, RESOURCE_STATUS.OFFLINE])
+        self.pod_event_with_func_type = SubscribeEvent(const.POD_EVENT, [
+            RESOURCE_STATUS.FAILED, RESOURCE_STATUS.ONLINE, RESOURCE_STATUS.DEGRADED, RESOURCE_STATUS.OFFLINE], [
+                NODE_FUNCTIONAL_TYPES.SERVER.value, NODE_FUNCTIONAL_TYPES.DATA.value])
+        self.pod_event_without_func_type = SubscribeEvent(const.POD_EVENT, [
+            RESOURCE_STATUS.FAILED, RESOURCE_STATUS.ONLINE, RESOURCE_STATUS.DEGRADED, RESOURCE_STATUS.OFFLINE])
+        self.disk_event_without_func_type = SubscribeEvent(const.DISK_EVENT, [
+            RESOURCE_STATUS.FAILED, RESOURCE_STATUS.ONLINE, RESOURCE_STATUS.OFFLINE])
+        self.pod_event_with_func_type_all = SubscribeEvent(const.POD_EVENT, [
+            RESOURCE_STATUS.FAILED, RESOURCE_STATUS.ONLINE, RESOURCE_STATUS.DEGRADED, RESOURCE_STATUS.OFFLINE],
+            [FUNCTIONAL_TYPES.ALL.value])
 
     def tearDown(self):
         pass
 
-    def test_subscriber(self):
-        self.event_manager.subscribe(self.component, [self.event])
-        self.event_manager.unsubscribe(self.component, [self.event])
+    def test_subscriber_with_functional_type(self):
+        # pod event with functional type
+        self.event_manager.subscribe(self.component, [self.pod_event_with_func_type])
+        #self.event_manager.unsubscribe(self.component, [self.pod_event_with_func_type])
+
+    def test_subscriber_without_functional_type(self):
+        # pod event without functional type
+        self.event_manager.subscribe(self.component, [self.pod_event_without_func_type])
+        #self.event_manager.unsubscribe(self.component, [self.pod_event_without_func_type])
+
+        # disk event without functional type
+        self.event_manager.subscribe(self.component, [self.disk_event_without_func_type])
+        #self.event_manager.unsubscribe(self.component, [self.disk_event_without_func_type])
+
+    def test_subscriber_with_functional_type_all(self):
+        # pod event without functional type
+        self.event_manager.subscribe(self.component, [self.pod_event_with_func_type_all])
+        #self.event_manager.unsubscribe(self.component, [self.pod_event_with_func_type_all])
+
```

Comment:
why unsubscribe function call in comment section , if its for future please add TODO line as well ,

Reply:
@Madhura-08  will enable these. I will add TODO.

Reply:
Done. @gauri-bhosale